doc/source/ray-core/examples/batch_prediction.ipynb

```diff
@@ -0,0 +1,347 @@
+    "\n",
+    "```{tip}\n",
+    "In general, actor-based batch prediction is preferred for higher efficiency. This is because the actor is stateful and it can reuse the expensive initialization across tasks.\n",
+    "```\n",
+    "\n",
+    "## Task-based batch prediction\n",
+    "\n",
+    "With Ray tasks, you can build a batch prediction program in this way:\n",
+    "1. Loads the model\n",
+    "2. Launches Ray tasks, with each taking in the model and a shard of input dataset\n",
+    "3. Each worker executes predictions on the assigned shard, and writes out results\n",
+    "\n",
+    "Let’s take NYC taxi data in 2009 for example. Suppose we have this simple model:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import pandas as pd\n",
+    "import numpy as np\n",
+    "\n",
+    "def load_model():\n",
+    "    # A dummy model.\n",
+    "    def model(batch: pd.DataFrame) -> pd.DataFrame:\n",
+    "        # Dummy payload so copying the model will actually copy some data\n",
+    "        # across nodes.\n",
+    "        model.payload = np.zeros(100_000_000)\n",
+    "        return pd.DataFrame({\"score\": batch[\"passenger_count\"] % 2 == 0})\n",
+    "    \n",
+    "    return model"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The dataset has 12 files (one for each month) so we can naturally have each Ray task to take one file. By taking the model and a shard of input dataset (i.e. a single file), we can define a Ray remote task for prediction:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import pyarrow.parquet as pq\n",
+    "import ray\n",
+    "\n",
+    "@ray.remote\n",
+    "def make_prediction(model, shard_path):\n",
+    "    df = pq.read_table(shard_path).to_pandas()\n",
+    "    result = model(df)\n",
+    "\n",
+    "    # Write out the prediction result.\n",
+    "    # NOTE: unless the driver will have to further process the\n",
+    "    # result (other than simply writing out to storage system),\n",
+    "    # writing out at remote task is recommended, as it can avoid\n",
+    "    # congesting or overloading the driver.\n",
+    "    # ...\n",
+    "\n",
+    "    # Here we just return the size about the result in this example.\n",
+    "    return len(result)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The driver launches all tasks for the entire input dataset. In order to not overload cluster and cause OOM, we use ray.wait() to limit the number of in-flight tasks, see details about this design pattern in {doc}`/ray-core/patterns/limit-tasks`."
```

Comment:
An alternative to this pattern is to estimate the memory usage per task and add a `memory=...` annotation in `ray.remote`, then the Ray scheduler will do the right thing -- we should at least point this out :)

Reply:
You can even do it on a per task basis with e.g. `make_prediction.options(memory=2 * file.size).remote(model_ref, file)` if the file size is available.

My guess is that this is the more convenient way to express this pattern -- it is both simpler and more robust (e.g. it doesn't depend on the total amount of memory in the cluster, whereas limiting parallelism with wait does).

Reply:
Good point, I added in.
It'll be more intuitive for cases where user can get a good estimate of in-memory data size. It's just not always the case, e.g. Parquet to Arrow decompression may have quite different inflations.